mlir/lib/Dialect/Vector/IR/VectorOps.cpp

```diff
@@ -2463,7 +2461,10 @@ static OpFoldResult foldFromElementsToElements(FromElementsOp fromElementsOp) {
 ///
 static OpFoldResult foldFromElementsToConstant(FromElementsOp fromElementsOp,
                                                ArrayRef<Attribute> elements) {
-  if (llvm::any_of(elements, [](Attribute attr) { return !attr; }))
+  // Check for null or poison attributes before any processing.
+  if (llvm::any_of(elements, [](Attribute attr) { 
+    return !attr || isa<ub::PoisonAttrInterface>(attr); 
```

Comment:
Long term, we should avoid hard-coding supported/unsupported types here. E.g., instead `DenseElementsAttr::get` could return an "empty" attribute instead of failing an assertion. But let's do that in a separate PR.

Reply:
Filed a follow up issue: https://github.com/llvm/llvm-project/issues/159609